qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -50,19 +52,49 @@ class FermionicOp(ParticleOp):
           - :math:`I - n = c c^\dagger`
           - Hole number
 
+    There are two types of label modes for :class:`FermionicOp`.
+    The label mode is automatically detected.
+
+    1. Dense Label (if underscore `_` does not exist in the label)
+
+    Dense labels are strings with allowed characters above.
+    This is similar to Qiskit's string-based representation of qubit operators.
+    For example,
+
+    .. code-block:: python
+
+        "+"
+        "II++N-IE"
+
+    are possible labels.
+
+    2. Sparse Label (if underscore `_` exists in the label)
+
+    A sparse label is a string consisting of a space-separated list of words.
+    Each word must look like :code:`[+-INE]_<index>`, where the :code:`<index>`
+    is a non-negative integer representing the index of the fermionic mode.
+    For example,
+
+    .. code-block:: python
+
+        "+_0"
+        "-_2"
+        "+_10 -_4 +_1 I_0 +_0"
+
+    are possible labels.
+    The :code:`index` must be in descending order.
```

Comment:
Is there a particular reason for this limitation?

Reply:
Yes. Because "+_1 -_0" is not equal to "-_0 +_1".

Reply:
Exactly. So why can't we allow the latter?

Reply:
Ah, I got what you said.

It is possible to implement it. But, I think it's non-trivial behavior to initialize an operator that is different from the input label.
Above case, `FermionicOp([('+-', -1)])` is generated.
If user wants to do that,  write explicitly
```
FermionicOp("I_1 -_0") @ FermionicOp("+_1")
```
for now. ("I_1" is not so cool, I want to solve this in the future...)

Reply:
Although for the purpose of generality it would be good to support any ordering, I think that respecting the ordering here is not bad in view of the commutators algebra that need to be developed further (in another PR). I believe that keeping the indices in descending order will make the reading of the operator easier. Solving the "I_1" would be nice, as @ikkoham pointed.

Reply:
What exactly needs to happen here to resolve `I_1`? Is that just required in order to make sure that the `register_length` is correct?

Reply:
We can choose:

1. Explicit: `register_length`
2. Implicit: automatically expansion to larger register_length operator

I prefer the explicit one from the perspective of the Zen of Python.